python/cudf/cudf/tests/test_index.py

```diff
@@ -1106,115 +1106,115 @@ def test_index_append_error(data, other):
+#     "data,other",
+#     [
+#         (
+#             pd.Index([1, 2, 3, 4, 5, 6]),
+#             [
+#                 pd.Index([1, 2, 3, 4, 5, 6]),
+#                 pd.Index([1, 2, 3, 4, 5, 6, 10]),
+#                 pd.Index([]),
+#             ],
+#         ),
+#         (
+#             pd.Index([]),
+#             [
+#                 pd.Index([1, 2, 3, 4, 5, 6]),
+#                 pd.Index([1, 2, 3, 4, 5, 6, 10]),
+#                 pd.Index([1, 4, 5, 6]),
+#             ],
+#         ),
+#         (
+#             pd.Index([10, 20, 30, 40, 50, 60]),
+#             [
+#                 pd.Index([10, 20, 30, 40, 50, 60]),
+#                 pd.Index([10, 20, 30]),
+#                 pd.Index([40, 50, 60]),
+#                 pd.Index([10, 60]),
+#                 pd.Index([60]),
+#             ],
+#         ),
+#         (
+#             pd.Index([]),
+#             [
+#                 pd.Index([10, 20, 30, 40, 50, 60]),
+#                 pd.Index([10, 20, 30]),
+#                 pd.Index([40, 50, 60]),
+#                 pd.Index([10, 60]),
+#                 pd.Index([60]),
+#             ],
+#         ),
+#         (
+#             pd.Index(["1", "2", "3", "4", "5", "6"]),
+#             [
+#                 pd.Index(["1", "2", "3", "4", "5", "6"]),
+#                 pd.Index(["1", "2", "3"]),
+#                 pd.Index(["6"]),
+#                 pd.Index(["1", "6"]),
+#             ],
+#         ),
+#         (
+#             pd.Index([]),
+#             [
+#                 pd.Index(["1", "2", "3", "4", "5", "6"]),
+#                 pd.Index(["1", "2", "3"]),
+#                 pd.Index(["6"]),
+#                 pd.Index(["1", "6"]),
+#             ],
+#         ),
+#         (
+#             pd.Index([1.0, 2.0, 3.0, 4.0, 5.0, 6.0]),
+#             [
+#                 pd.Index([1.0, 2.0, 3.0, 4.0, 5.0, 6.0]),
+#                 pd.Index([1.0, 6.0]),
+#                 pd.Index([]),
+#                 pd.Index([6.0]),
+#             ],
+#         ),
+#         (
+#             pd.Index([]),
+#             [
+#                 pd.Index([1.0, 2.0, 3.0, 4.0, 5.0, 6.0]),
+#                 pd.Index([1.0, 6.0]),
+#                 pd.Index([1.0, 2.0, 6.0]),
+#                 pd.Index([6.0]),
+#             ],
+#         ),
+#         (
+#             pd.Index(["a"]),
+#             [
+#                 pd.Index(["a"]),
+#                 pd.Index(["a", "b", "c"]),
+#                 pd.Index(["c"]),
+#                 pd.Index(["d"]),
+#                 pd.Index(["ae", "hello", "world"]),
+#             ],
+#         ),
+#         (
+#             pd.Index([]),
+#             [
+#                 pd.Index(["a"]),
+#                 pd.Index(["a", "b", "c"]),
+#                 pd.Index(["c"]),
+#                 pd.Index(["d"]),
+#                 pd.Index(["ae", "hello", "world"]),
+#                 pd.Index([]),
+#             ],
+#         ),
+#     ],
+# )
+# def test_index_append_list(data, other):
```

Comment:
Is this test supposed to be commented out?

Reply:
@mroeschke i commented it out because it fails, which is in regards to my question [here](https://github.com/rapidsai/cudf/pull/10389#discussion_r986133273) :

You suggested to do a top-level check of `if not isinstance(other, BaseIndex)`, does this mean we don't want to consider cases where other can be  `is_list_like` obj like pandas does?

Reply:
You can ignore my prior suggestion and just keep the original code as it was (and have `other` accept list like objects).